raiden/utils/filters.py

```diff
@@ -89,3 +96,39 @@ def decode_event(abi: Dict, log: Dict):
     }
     event_abi = topic_to_event_abi[event_id]
     return get_event_data(event_abi, log)
+
+
+class StatelessFilter(LogFilter):
+    _last_block: int = -1
+
+    def __init__(self, web3: Web3, filter_params: dict):
+        super().__init__(web3, filter_id=None)
+        self.filter_params = filter_params
+
+    def get_new_entries(self):
+        filter_params = self.filter_params.copy()
+        filter_params['fromBlock'] = max(
+            filter_params.get('fromBlock', 0),
+            self._last_block + 1,
+        )
+        if self.filter_params.get('toBlock') in ('latest', 'pending'):
+            filter_params['toBlock'] = self.web3.eth.blockNumber
```

Comment:
I would maybe add a lock (or just an assert to test the assumption) as we may never be sure that other greenlet will access the filter during the context switch that can happen while calling `web3.eth.blockNumber` ot `web3.eth.getLogs`.